Element84/filmdrop-ui
# adding draw boundary feature for search

**Related Issue(s):**

- NA

**Proposed Changes:**

> NOTE: part of work contributing to adding a work in progress feature

1. hide advanced search options behind `VITE_ADVANCED_SEARCH_ENABLED` flag from config
2. move geosearch to be below zoom controls and render as a closed button
3. add draw boundary feature to allow user to draw polygon on map
4. if polygon drawn, use as search intersects param instead of map viewport bbox

BOUNS: fixed some a11y linting errors in touched files
BOUNS: fixed some synk library vulnerabilities (pinned semver)

### **To Test:**

**confirm UI matches any mocks available to reviewer (see internal ticket for mocks)**

**unit tests pass 🍏**

**test config flag**

- set `VITE_ADVANCED_SEARCH_ENABLED` to false in config
- load app
- confirm auto search toggle renders in UI
- confirm advanced options button does not render in UI
- set `VITE_ADVANCED_SEARCH_ENABLED` to true
- reload app
- - confirm auto search toggle does not render in UI
- confirm advanced options button renders in UI

**test geosearch changes**
- set `VITE_ADVANCED_SEARCH_ENABLED` to true OR false (should render same in both)
- load app
- confirm geosearch control is below zoom controls in top left of map
- confirm geosearch control is collapsed by default 
- while zoomed out a low level, click search button
- confirm `Images are not visible at this zoom level.` message now is positioned to be aligned with top of zoom control (instead of retaining padding from when it was below opened geosearch)
 
 **test draw feature**
- set `VITE_ADVANCED_SEARCH_ENABLED` to true
- load app
- click `advanced` toggle below search button
- confirm advanced options render
- confirm `draw boundary` and `upload geojson` buttons are enabled
- confirm `clear` button is disabled
- click `draw boundary` button
- confirm search tools overlay renders and search params are not editable while drawing
- confirm `draw search boundary` message shows
- confirm map cursor changes to be draw tool
- draw polygon
- click first point again to complete polygon
- confirm `draw search boundary` message no longer renders
- confirm advanced options close
- confirm search tools overlay closes and search params are editable again
- confirm new polygon is now present on the map
- click `advanced` toggle below search button again
- confirm `draw boundary` and `upload geojson` buttons are disabed
- confirm `clear` button is enabled
- click clear button
- confirm polygon is cleared from the map
- confirm advanced options close
- click `advanced` toggle below search button again
- confirm `draw boundary` and `upload geojson` buttons are enabled
- confirm `clear` button is disabled

**test search with drawn polygon**
- set `VITE_ADVANCED_SEARCH_ENABLED` to true
- load app
- click `advanced` toggle below search button
- click `draw boundary` button
- draw polygon
- click first point again to complete polygon
- click search button
- confirm search results are only returned that intersect polygon
- change collection and other search params
- click search button again
- confirm search results are still only returned that intersect polygon
- click mosaic toggle
- confirm mosaic rendered also only covers area around the intersect polygon
- click back to scene mode
- click `advanced` toggle below search button again
- click clear
- click search button again
- confirm new search uses full viewport as search bounds

**test messages show/hide correctly when drawing**
- set `VITE_ADVANCED_SEARCH_ENABLED` to true
- load app
- search so that results are showing on the map
- confirm message for `Showing Aggregated Results` or `Showing x of y scenes` message is displayed
- click `advanced` toggle below search button
- click `draw boundary` button
- confirm `draw search boundary` message shows
- confirm message for `Showing Aggregated Results` or `Showing x of y scenes` message is no longer displayed
- draw polygon
- click first point again to complete polygon
- confirm `draw search boundary` message no longer shows
- confirm message for `Showing Aggregated Results` or `Showing x of y scenes` message is displayed again

**test cancel drawing**
- set `VITE_ADVANCED_SEARCH_ENABLED` to true
- load app
- click `advanced` toggle below search button
- click `draw boundary` button
- click cancel button in `draw search boundary` message
- confirm `draw search boundary` message no longer renders
- confirm advanced options close
- confirm search tools overlay closes and search params are editable again
- confirm polygon not added to map

**PR Checklist:**
